ui/selectors/selectors.js

```diff
@@ -1573,6 +1589,48 @@ export const getMemoizedUnapprovedTypedMessages = createDeepEqualSelector(
   (unapprovedTypedMessages) => unapprovedTypedMessages,
 );
 
+/**
+ * Get the display name for an address.
+ * This selector will look into the internal accounts and address book to find a display name for the address.
+ *
+ * @param _state - The Redux state object.
+ * @param {string} address - The address to get the display name for in a CAIP-10 format.
+ * @returns {string} The display name for the address.
+ */
+export const getAddressDisplayName = createDeepEqualSelector(
```

Comment:
I don' think the memoization will be effective if we pass in the entire raw state. Wondering how we can structure this to only reference the state we need 🤔